fireplace/actions.py

```diff
@@ -705,9 +707,24 @@ class Discover(TargetedAction):
 	ARGS = ("TARGET", "CARDS")
 
 	def get_target_args(self, source, target):
+		# use hero class for Discover if not neutral (eg. Ragnaros)
+		if target.hero.data.card_class != CardClass.NEUTRAL:
+			discoverclass = target.hero.data.card_class
+
+		# use card class for neutral hero classes
+		elif source.data.card_class != CardClass.NEUTRAL:
+			discoverclass = source.data.card_class
+
+		# use random class for neutral hero classes with neutral cards
```

Comment:
Yes :) This ruleset is on the wiki; spod and I played a whole bunch of games and confirmed that is how it appears to work.